python/packages/autogen-agentchat/src/autogen_agentchat/agents/_society_of_mind_agent.py

```diff
@@ -143,7 +174,19 @@ async def on_messages_stream(
         # Run the team of agents.
         result: TaskResult | None = None
         inner_messages: List[BaseAgentEvent | BaseChatMessage] = []
+        model_context = self._model_context
         count = 0
+
+        prev_content = await model_context.get_messages()
+        if len(prev_content) > 0:
+            prev_message = HandoffMessage(
+                content="relevant previous messages",
+                source=self.name,
+                target="",
+                context=prev_content,
+            )
+            task = [prev_message] + task
```

Comment:
We need to address the case when `task` list can be empty. See: https://github.com/microsoft/autogen/discussions/6188